keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/DeliveryArtifact.kt

```diff
@@ -30,5 +30,15 @@ abstract class DeliveryArtifact {
   abstract val reference: String // friendly reference to use within a delivery config
   abstract val deliveryConfigName: String? // the delivery config this artifact is a part of
   open val statuses: Set<ArtifactStatus> = emptySet()
+
+  fun toPublishedArtifact(version: String, status: ArtifactStatus? = null) =
+    PublishedArtifact(
+      name = name,
+      type = type,
+      reference = reference,
+      version = version,
+      metadata = mapOf("releaseStatus" to status)
+    ).normalized()
```

Comment:
I'm catching up on recent artifact work. Can you explain to me like I'm 5 the difference between a `DeliveryArtifact` and a `PublishedArtifact`? Does the latter represent a specific instance of the former with a version and associated metadata? That seems to be the case, but I feel like `PublishedArtifact` is maybe not the best name. Neither is `DeliveryArtifact` for that matter, but one thing at a time.

Reply:
Yep, that's exactly it. `DeliveryArtifact` really just has some metadata (type, name, etc.) that allows us to find and sort the actual artifacts. I've been struggling with these names too. A couple of variants I thought of for `PublishedArtifact` were `PublishedArtifactVersion` or just `ArtifactVersion`. Does any of those sound better to you?

Reply:
@robfletcher I'm merging this as is, but let me know. I can change the name in a follow-up PR.